tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,65 @@
+import json
+import subprocess
+import time
+
+
+def test_dcgm_exporter_endpoint():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "localhost:9400/metrics"
+    timeout = 60  # seconds
+    start_time = time.time()
+
+    def query_endpoint():
+        return subprocess.run(["curl", endpoint], text=True, capture_output=True)
+
+    while (result := query_endpoint()).returncode != 0 or not result.stdout.strip():
+        if time.time() - start_time > timeout:
+            assert False, f"Failed to reach '{endpoint}' of the dcgm-exporter service"
+        time.sleep(5)
+
+    assert "DCGM_FI_DRIVER_VERSION" in result.stdout, "No dcgm exported metrics found"
```

Comment:
I guess this will fail in a VM because it's not expected to have metrics at all.

My suggestion is to maybe run a `lshw` to see if there is an NVIDIA GPU and in that case test that exist some metrics, otherwise just check if the response is 200, otherwise this will fail to the CI.

Detecting to check different things will make easier to test on both local and CI